pyroSAR/snap/util.py

```diff
@@ -231,6 +241,25 @@ def geocode(infile, outdir, t_srs=4326, tr=20, polarizations='all', shapefile=No
         cal.parameters['outputBetaBand'] = False
         cal.parameters['outputGammaBand'] = True
     ############################################
+    # speckle filtering node configuration
+    speckleFilter_options = ['Boxcar',
+                             'Median',
+                             'Frost',
+                             'Gamma Map',
+                             'Refined Lee',
+                             'Lee',
+                             'Lee Sigma']
+    
+    if speckleFilter:
+        message = '{0} must be one of the following:\n- {1}'
+        if speckleFilter not in speckleFilter_options:
+            raise ValueError(message.format('speckleFilter', '\n- '.join(speckleFilter_options)))
+        sf = parse_node('Speckle-Filter')
+        workflow.insert_node(sf, before='Terrain-Flattening')
```

Comment:
please make use of the `last` variable in this line instead of the hard-coded 'Terrain-Flattening'